examples/tictactoe/programs/tictactoe/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+use anchor_lang::prelude::*;
+use std::str::FromStr;
+
+const BOARD_ITEM_FREE: u8 = 0; // Free slot
+const BOARD_ITEM_X: u8 = 1; // Player X
+const BOARD_ITEM_O: u8 = 2; // Player O
+
+/// Game State
+/// 0 - Waiting
+/// 1 - XMove
+/// 2 - OMove
+/// 3 - XWon
+/// 4 - OWon
+/// 5 - Draw
+
+#[program]
+pub mod tictactoe {
+    use super::*;
+
+    #[access_control(Initializedashboard::accounts(&ctx))]
```

Comment:
This `Accounts::accounts` pattern has been deprecated in favor newer syntax.

It would be more idiomatic to remove all `Accounts::accounts()` methods and do the account validation via constraints inside `#[derive(Accounts)]`. 

To access instruction data from there, use 

```rust
#[derive(Accounts)]
#[instruction(data: u64, more_data: String)]
pub struct MyAccounts {
...
}
```

For `#[access_control]` that is not related to account validation, a separately named function would be great--i.e. something more specific than `Initializedashboard::accounts`

Reply:
Fixed the first constraints for the first 3 instructions. third one seemed complicated. Isn't it better to have a function for it ?

Reply:
Sounds good to me!

Reply:
Okay so I leave it with #[acess_control] ?